src/shared/utils/challenge-listing/filter.js

```diff
@@ -113,7 +113,9 @@ function filterBySubtracks(challenge, state) {
    * it probably does not work in all cases. It should be double-checked,
    * why challenge subtracks in challenge objects are different from those
    * return from the API as the list of possible subtracks. */
-  const filterSubtracks = state.subtracks.map(item =>
+  const subtracks = Array.isArray(state.subtracks) ? state.subtracks :
+    Object.keys(state.subtracks).map(key => state.subtracks[key]);
+  const filterSubtracks = subtracks.map(item =>
```

Comment:
Especially here. It is just documented in the module's head comment that `tags` and `subtracks` must be arrays. So if it gets called with object, it is an error to be fixed somewhere else.